lib/puppet/provider/ec2_securitygroup/v2.rb

```diff
@@ -131,11 +131,15 @@ def authorize_ingress(new_rules, existing_rules=[])
     to_create.reject(&:nil?).each do |rule|
       if rule.key? 'security_group'
         source_group_name = rule['security_group']
-        group_response = ec2.describe_security_groups(filters: [
-          {name: 'group-name', values: [source_group_name]},
-        ])
+        filters = [ {name: 'group-name', values: [source_group_name]} ]
+        if vpc_only_account?
+          response = ec2.describe_security_groups(group_ids: [@property_hash[:id]])
+          vpc_id = response.data.security_groups.first.vpc_id
+          filters.push( {name: 'vpc-id', values: [vpc_id]} )
+        end
+        group_response = ec2.describe_security_groups(filters: filters)
         fail("No groups found called #{source_group_name}") if group_response.data.security_groups.count == 0
-        source_group_id = group_response.data.security_groups.first.group_id
+        source_group_id = group_response.data.security_groups.find { |g| g.group_name == source_group_name }.group_id
```

Comment:
Note that this was added based on what the AWS console shows when filtering - i.e. it does a substring match and returns `default_elb_XXX` when passing in `default` as the filter.

However, I've verified that in code / other environments, this is an exact string match as we've expected.

I modified code using the Ruby SDK to emit the list of security groups with `{:name=>"group-name", :values=>["default"]}` used as a filter, and only those that exactly match `default` are returned.

I also used the AWS CLI with `--filters "Name=group-name,Values=default"` to verify that only exact matches to `default` are returned.

So substring matched group names is not exactly the problem.  However, I do see two matches come back in our code -- but they're the same thing.  So I suspect something else is afoot.